### ISSUE TITLE
CVC4 incorrect model on UF formula with distinct operator

### ISSUE THREAD

**Issue by muchang** [CVC4/CVC4#3527]
Hi,
For this formula
```
(declare-fun f (Int) Bool)
(declare-fun g (Int) Bool)
(assert (distinct f g))
(check-sat)
(get-model)
```
CVC4 gives an incorrect model:
```
(model
(define-fun f ((BOUND_VARIABLE_294 Int)) Bool false)
(define-fun g ((BOUND_VARIABLE_294 Int)) Bool false)
)
```

OS: Ubuntu 18.04
Revision: f3d40cc

**Comment by ajreynol** [comment]
This file involves higher-order reasoning, you should use `--uf-ho` for problems with higher-order reasoning. With this option, we produce this model:
```
sat
(model
(define-fun f ((BOUND_VARIABLE_313 Int)) Bool false)
(define-fun g ((BOUND_VARIABLE_313 Int)) Bool (= BOUND_VARIABLE_313 0))
)
```
The issue here is we are not throwing an exception when higher-order constraints are used without `--uf-ho`.  We are in the process of fixing this.

**Comment by ajreynol** [comment]
Fixed in #3551.
